packages/devtools_app/lib/src/screens/memory/framework/memory_controller.dart

```diff
@@ -65,26 +75,82 @@ class MemoryController extends DisposableController
     profile.dispose();
   }
 
-  DiffPaneController _createDiffController() =>
-      DiffPaneController(HeapGraphLoaderRuntime(chart.memoryTimeline));
+  Future<void> _init(
+    @visibleForTesting DiffPaneController? connectedDiff,
+    @visibleForTesting ProfilePaneController? connectedProfile,
+  ) async {
+    assert(!isInitialized.value);
+    switch (_mode) {
+      case DevToolsMode.disconnected:
+        throw StateError('Memory screen does not support disconnected mode.');
+      case DevToolsMode.connected:
+        _initializeData(
+          diffPaneController: connectedDiff,
+          profilePaneController: connectedProfile,
+        );
+      case DevToolsMode.offlineData:
+        assert(connectedDiff == null && connectedProfile == null);
+        await maybeLoadOfflineData(
+          PerformanceScreen.id,
+          createData: (json) => OfflineMemoryData.parse(json),
+          shouldLoad: (data) => !data.isEmpty,
+        );
+        // If shouldLoad returns false, previous line is noop, so data should be initialized.
```

Comment:
nit: can we make this comment more clear 
"[maybeLoadOfflineData] will be a noop if we should not load offline data for the memory screen, so ensure we still call [_initializedData] if it has not been called."

Reply:
not relevant any more